packages/grafana-runtime/src/services/live.ts

```diff
@@ -70,3 +75,15 @@ export const setGrafanaLiveSrv = (instance: GrafanaLiveSrv) => {
  * @public
  */
 export const getGrafanaLiveSrv = (): GrafanaLiveSrv => singletonInstance;
+
+/**
+ * Returns a promise that resolves with the {@link GrafanaLiveSrv} once it has
+ * connected
+ *
+ * @experimental
+ * @public
+ */
+export const getConnectedLiveSrv = async () => {
+  await singletonInstance.waitUntilConnected();
+  return singletonInstance;
+};
```

Comment:
Looking at how we use the currently it does not look like we need two getLiveSrv functions. The connect happens in the constructor. So everyone who is interested in the LiveSrv should probably use getConnectedLiveSrv, so I would make this the default behavior. and only have a single getGrafanaLiveSrv 

But I also question a bit the need for this in the first place. Can't we make initChannel / getChannelStream / publish handle this connection wait internally?

Reply:
but having this as default could be a start

Reply:
This is fair actually, let me make some changes

Reply:
I agree with Torkel, as a consumer I'm only interested in getting a stream to subscribe/publish too.